cmake/modules/Logging.cmake

```diff
@@ -35,21 +26,30 @@ if("${USE_LIBBACKTRACE}" STREQUAL "AUTO")
   message(STATUS "Autoset: USE_LIBBACKTRACE=" ${USE_LIBBACKTRACE} " in " ${CMAKE_SYSTEM_NAME})
 endif()
 
+if (USE_LIBBACKTRACE)
+    set(current_dir ${CMAKE_CURRENT_LIST_DIR})
+endif ()
 
-if(USE_LIBBACKTRACE)
-  message(STATUS "Building with libbacktrace...")
-  include(cmake/libs/Libbacktrace.cmake)
-  target_link_libraries(tvm PRIVATE libbacktrace)
-  target_link_libraries(tvm_runtime PRIVATE libbacktrace)
-  add_dependencies(tvm_runtime_objs libbacktrace)
-  # pre 3.12 versions of cmake cannot propagate include directories from imported targets so we set them manually
-  target_include_directories(tvm PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/libbacktrace/include")
-  target_include_directories(tvm_objs PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/libbacktrace/include")
-  target_include_directories(tvm_runtime PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/libbacktrace/include")
-  target_include_directories(tvm_runtime_objs PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/libbacktrace/include")
-  target_compile_definitions(tvm_objs PRIVATE TVM_USE_LIBBACKTRACE=1)
-  target_compile_definitions(tvm_runtime_objs PRIVATE TVM_USE_LIBBACKTRACE=1)
-else()
-  target_compile_definitions(tvm_objs PRIVATE TVM_USE_LIBBACKTRACE=0)
-  target_compile_definitions(tvm_runtime_objs PRIVATE TVM_USE_LIBBACKTRACE=0)
-endif()
+set(backtrace_defined OFF CACHE BOOL "backtrace defined indicator")
+
+macro(use_backtrace package)
+  if (USE_LIBBACKTRACE)
+    message(STATUS "Building with libbacktrace...")
+    if (NOT backtrace_defined)
+        include(${current_dir}/../libs/Libbacktrace.cmake)
+        set(backtrace_defined ON CACHE BOOL "backtrace defined indicator" FORCE)
+    endif()
+    
+    target_link_libraries(${package} PRIVATE backtrace)
+    if ("${package}" STREQUAL "tvm_runtime")
+        target_link_libraries(tvm_runtime backtrace)
```

Comment:
A little confused as to why `tvm_runtime` gets special behavior here. It does not match the previous behavior. Could you explain?